src/SourceIndexServer/Models/ResultsHtmlGenerator.cs

```diff
@@ -208,9 +209,13 @@ private void AppendAffiliateLinks(Query query)
             var term = query.OriginalString;
             term = Markup.UrlEncodeAndHtmlEscape(term);
             WriteLine("<ul>");
-            AppendAffiliateLink("http://www.bing.com/search?q=" + term);
-            AppendAffiliateLink("http://social.msdn.microsoft.com/Search/en-US?query=" + term);
-            AppendAffiliateLink("http://stackoverflow.com/search?q=" + term);
+            // Read the AffiliateLinks file and display links one by one.
+            
+            foreach (string line in File.ReadLines("..\\AffiliateLinks.txt"))

```

Comment:
I've just realized another problem with this approach - this is going to hit the disk on every request, unnecessarily. Could you read the lines once when the service starts and never read the file again? Just have a static field and store the affiliate links there.